### PR TITLE
NEW Adding a stack more unit tests for logging and some form fields

### PR DESCRIPTION
**Pull request by robbieaverill**
This is adding more tests for currently convered parts of the codebase.

See commit messages for details.

### CODE REVIEW

**Comment by robbieaverill** [comment]
@silverstripe/core-team someone feel like reviewing this before it gets any bigger? 😛

File: src/ORM/ListDecorator.php [hunk 1]
```diff
@@ -173,8 +173,7 @@ public function reverse()
      */
     public function sort()
```
[the remark this answers is not in the document]
Comment (robbieaverill) [reply src/ORM/ListDecorator.php]: That would change the method signature though, need to preserve it for semver

File: tests/php/Forms/CurrencyField_DisabledTest.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?php
+
+namespace SilverStripe\Forms\Tests;
+
+use SilverStripe\Dev\SapphireTest;
+use SilverStripe\Forms\CurrencyField_Disabled;
+use SilverStripe\ORM\FieldType\DBCurrency;
+
+class CurrencyField_DisabledTest extends SapphireTest
```
Comment (ScopeyNZ) [inline tests/php/Forms/CurrencyField_DisabledTest.php]: Ah there's actually a bunch of these 😴

Comment (robbieaverill) [reply tests/php/Forms/CurrencyField_DisabledTest.php]: I think I've caught them all now, thanks!

**Comment by ScopeyNZ** [comment]
Nice effort! 🙂

**Pull request merged**